drivers/sensor/apds9960/apds9960.c

```diff
@@ -0,0 +1,489 @@
+{
+	struct apds9960_data *data = dev->driver_data;
+	int temp = 0;
+	u8_t val_low;
+	u8_t val_high;
+
+	/* Break 16-bit threshold into 2 8-bit values */
+	val_low = threshold & 0x00FF;
+	val_high = (threshold & 0xFF00) >> 8;
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_INT_AIHTL_REG, val_low);
+	if (temp < 0) {
+		SYS_LOG_DBG(" Failed");
+		return temp;
+	}
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_INT_AIHTH_REG, val_high);
+	if (temp < 0) {
+		SYS_LOG_DBG(" Failed");
+		return temp;
+	}
+
+	return 0;
+}
+
+static int apds9960_proxy_setup(struct device *dev, int gain)
+{
+	struct apds9960_data *data = dev->driver_data;
+	int temp = 0;
+
+	/* Power ON */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, BIT(0), APDS9960_POWER_ON);
+	if (temp < 0) {
+		SYS_LOG_DBG("Power on bit not set.");
+		return temp;
+	}
+
+	/* ADC value */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ATIME_REG, APDS9960_ATIME_WRTIE, APDS9960_ADC_VALUE);
+	if (temp < 0) {
+		SYS_LOG_DBG("ADC bits are not written");
+		return temp;
+	}
+
+	/* proxy Gain */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONTROL_REG, APDS9960_BITS_32,
+		(gain & APDS9960_PGAIN_8X));
+
+	if (temp < 0) {
+		SYS_LOG_DBG("proxy Gain is not set");
+		return temp;
+	}
+
+	/* Enable proxy */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, APDS9960_BITS_20, APDS9960_PROXY_ON);
+	if (temp < 0) {
+		SYS_LOG_DBG("Proxy is not enabled");
+		return temp;
+	}
+
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, BIT(5), 0x00);
+	if (temp < 0) {
+		SYS_LOG_DBG("Proxy interrupt is not enabled");
+		return temp;
+	}
+
+	return temp;
+}
+
+static int apds9960_ambient_setup(struct device *dev, int gain)
+{
+	struct apds9960_data *data = dev->driver_data;
+	int temp = 0;
+
+	/* Power ON */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, BIT(0), APDS9960_POWER_ON);
+	if (temp < 0) {
+		SYS_LOG_DBG("Power on bit not set.");
+		return -EIO;
+	}
+
+	/* ADC value */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ATIME_REG, APDS9960_ATIME_WRTIE, APDS9960_ADC_VALUE);
+	if (temp < 0) {
+		SYS_LOG_DBG("ADC bits are not written");
+		return temp;
+	}
+
+	/* ALE Gain */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONTROL_REG, APDS9960_BITS_10,
+		(gain & APDS9960_AGAIN_64X));
+	if (temp < 0) {
+		SYS_LOG_DBG("Ambient Gain is not set");
+		return temp;
+	}
+
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, BIT(4), 0x00);
+	if (temp < 0) {
+		return temp;
+	}
+
+	/* Enable ALE */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, APDS9960_BITS_10, APDS9960_RGB_ON);
```

Comment:
Later when we introduce interrupt we need to write this bit on some conditions( #ifdef) so let not combine this as of now. What do you think is it ready to merge ?